src/navigation/RootNavigator.tsx

```diff
@@ -11,12 +13,30 @@ import {
   SignInScreen,
   SignUpScreen,
 } from '~screens'
+import { registerForPushNotificationsAsync } from '~services'
 
 const { Navigator, Screen, Group } = createStackNavigator<RootStackParamList>()
 
 export const RootNavigator: FC = () => {
   const { t } = useTranslation()
   const { isSignedIn } = useAuth()
+  const { notification, setNotification } = useNotificationContext()
+
+  useEffect(() => {
+    const initNotifications = async () => {
+      await registerForPushNotificationsAsync()
+    }
+    initNotifications()
+  }, [])
+
+  useEffect(() => {
+    if (notification) {
+      Alert.alert('Notification', JSON.stringify(notification), [
+        { text: 'Cancel', onPress: () => setNotification(undefined), style: 'cancel' },
+        { text: 'Ok', onPress: () => setNotification(undefined) },
+      ])
+    }
+  }, [notification, setNotification])
```

Comment:
Maybe move it to seperate file, for example `useNotificationsSetup`, it will be easier to maintain it in the future